alchemist-graphql/src/jvmTest/kotlin/it/unibo/alchemist/boundary/TestingEnvironments.kt

```diff
@@ -15,19 +15,22 @@ import it.unibo.alchemist.model.EuclideanEnvironment
 import it.unibo.alchemist.model.Position
 import it.unibo.alchemist.model.geometry.Vector
 import it.unibo.alchemist.test.loadYamlSimulation
+import it.unibo.alchemist.test.startSimulation
 import java.io.File
 
 object TestingEnvironments {
-    fun <T, P> graphqlTestEnvironmnets(): Set<EuclideanEnvironment<T, P>> where P : Position<P>, P : Vector<P> =
+    fun <T, P> graphqlTestEnvironments(): Sequence<EuclideanEnvironment<T, P>> where P : Position<P>, P : Vector<P> =
         this::class.java.classLoader.getResource("yaml")?.path?.let { path ->
-            File(path).listFiles()?.map {
-                loadYamlSimulation<T, P>("yaml/${it.name}")
-            }?.toSet()
+            File(path).listFiles()?.asSequence()?.distinct()?.map {
+                val env = loadYamlSimulation<T, P>("yaml/${it.name}")
+                env.startSimulation()
+                env
+            }
         }.orEmpty()
```

Comment:
I would fail on an empty sequence, if there is at least one test

Reply:
If the sequence is empty (i.e. no yaml files were found in the `resource/yaml` folder),  the [`graphqlTestEnvironments` test](https://github.com/AlchemistSimulator/Alchemist/blob/9d4287d8ff8e4ec6ff083193dc2c3f40ab8a0cb0/alchemist-graphql/src/jvmTest/kotlin/it/unibo/alchemist/boundary/TestingEnvironments.kt#L34) will fail. I do not know if I understood correctly your point.